Assets/Global/Scripts/Player/Player.cs

```diff
@@ -121,26 +124,35 @@ public void OnCollisionEnter(Collision collision)
     }
 
     public void OnCollisionExit(Collision collision) { }
-    
+
     private void RaycastDown()
     {
         groundCheckDistance = rb.GetComponent<Collider>().bounds.extents.y;
-        RaycastHit hit;
-        Vector3 raycastPosition = new Vector3(rb.position.x, rb.position.y, rb.position.z);
-        if (Physics.Raycast(raycastPosition, Vector3.down, out hit, groundCheckDistance))
+        Vector3[] raycastOffsets = new Vector3[]
+        {
+            Vector3.zero, 
+            new Vector3(0, 0, rb.GetComponent<Collider>().bounds.extents.z), 
+            new Vector3(0, 0, -rb.GetComponent<Collider>().bounds.extents.z),
+            new Vector3(rb.GetComponent<Collider>().bounds.extents.x, 0, 0), 
+            new Vector3(-rb.GetComponent<Collider>().bounds.extents.x,0,0) ,
+        };
+
+        foreach (Vector3 offset in raycastOffsets)
         {
-            if (!(hit.collider.gameObject.CompareTag("Player")))
+            Vector3 raycastPosition = rb.position + offset;
+            if (Physics.Raycast(raycastPosition,Vector3.down,out RaycastHit hit,groundCheckDistance))
             {
-                if (Vector3.Angle(Vector3.up, hit.normal) < degreesToRotate)
+                if (!hit.collider.gameObject.CompareTag("Player"))
```

Comment:
we could get rid of this check if we utilize the Unity layers. 
Passing a Layer Mask to the Raycast to tell it to ignore a specific layer that we give the player, instead of checking after the fact.

Reply:
Yea, that would be nice. But should this PR wait on that? Or implement that? Or can it go thourgh and just do that later in a seperate task?